File: src/sbt-test/sbt-reactive-app/detection/lagom-java-impl/src/main/java/right/SimpleServiceImpl.java
package right;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import java.util.concurrent.CompletableFuture;

public class SimpleServiceImpl implements SimpleService {

  @Override
  public ServiceCall<NotUsed, String> simple(String input) {
    return req -> {
      CompletableFuture<String> result = CompletableFuture.completedFuture(input.toUpperCase());
      return result;
    };
  }
}

File: src/sbt-test/sbt-reactive-app/detection/lagom-scala-impl/src/main/scala/left/SimpleServiceImpl.scala
package left

import akka.NotUsed
import com.lightbend.lagom.scaladsl.api.ServiceCall

import scala.concurrent.Future

class SimpleServiceImpl extends SimpleService {
  override def simple(input: String): ServiceCall[NotUsed, String] = ServiceCall { _ =>
    Future.successful(input.toUpperCase)
  }
}
